lib/matplotlib/backends/backend_macosx.py

```diff
@@ -248,5 +211,19 @@ def set_message(self, message):
 #
 ########################################################################
 
-FigureCanvas = FigureCanvasMac
-FigureManager = FigureManagerMac
+@_Backend.export
+class _BackendMac(_Backend):
+    FigureCanvas = FigureCanvasMac
+    FigureManager = FigureManagerMac
+
+    def trigger_manager_draw(manager):
+        # For performance reasons, we don't want to redraw the figure after
+        # each draw command. Instead, we mark the figure as invalid, so that it
+        # will be redrawn as soon as the event loop resumes via PyOS_InputHook.
+        # This function should be called after each draw event, even if
+        # matplotlib is not running interactively.
```

Comment:
This comment, taken from the original, has me puzzled.  It seems to be contradicted by the code, which does nothing unless `is_interactive()` returns True.  This is not changed by your refactor, so if the comment is misleading now, then it was also misleading before; or I am just missing its point in both cases.